### ISSUE TITLE
Cash transport vans trying to enter a Cargo Hub fail to enter the building and spin in front of it

### ISSUE THREAD

**Issue by Rhacoal** [CitiesSkylinesMods/TMPE#1719]
### Describe the problem

Cash transport vans bounding for a **Cargo Service Point** may find its path through the **Cargo Hub** where they take a train to the service point, as the screenshot shows.

After such a path is found, a cash transport van will try to enter the Cargo Hub and fail to do so. They just keep spinning in front of the Cargo Hub and block the traffic behind.

With TMPE disabled, such a path will not be established.

With Cargo Hub replaced with Cargo Harbor, such a path will not be established.

### Steps to reproduce

1. Build a Cargo Hub
2. Build a pedestrian district with commercial buildings
3. Build a bank near the Cargo Hub and a cargo train station near the cargo service point
4. Waiting for a cash transport van targeting the service point to spawn

### Log files

`output_log.txt` and `TMPE.log`

https://ufile.io/f/j7xpq

### Savegame

[cash_van_spinning_bug_report_tmpe](https://steamcommunity.com/sharedfiles/filedetails/?edit=true&id=2916345276)

### Screenshots

![Path to the Cargo Hub](https://user-images.githubusercontent.com/10363029/211791342-9a58227e-786e-4e73-99ab-90d1d94a5982.png)

![Spinning vans](https://user-images.githubusercontent.com/10363029/211791363-f67ffa51-a16d-4ff5-ac79-aea0374d886b.png)

### Notes or questions

Game version: 1.16.0-f3
TMPE verson: 11.7.3.1 STABLE

**Comment by krzychu124** [comment]
Hmm, it might be vanilla bug, because it allows bank vans to use train, ship or plane to move cargo (cash in this case) but bank van AI internal item class service is not supported by the other side where normally should be despawned (cargo hub/port/airport buildings) 
> With TMPE disabled, such a path will not be established.

How did you test that? The proper way is to always run exactly the same savegame, once with and once without the mod, so the starting point (simulation state) will be the same in both runs. Transfer offer matching is more or less random so you might just get unlucky and vehicle selected different building or because of slightly different pathfinding calculations, using hub was not the optimal way.

I'll test and investigate your savegame later today anyways. Maybe it's something I can temporarily fix at least or prepare more detailed bug report.

**Comment by Rhacoal** [comment]
> How did you test that? The proper way is to always run exactly the same savegame, once with and once without the mod, so the starting point (simulation state) will be the same in both runs.

Yes, I tested it the same way as you said. I saved the game once and loaded it with and without TMPE. Without the mod, cash vans no longer enters train stations or cargo hubs.

I was not sure whether it was a compatibility issue, so I created a new savegame with only Unlimited Money, Unlock All and TMPE (with Unified UI and some other dependencies). After some time, I managed to reproduced the problem. But this time, when I disabled TMPE, the bug existed.

The savegame is here:
[tmpe_bug_report_2](https://steamcommunity.com/sharedfiles/filedetails/?id=2917746166)

![spinning vans 2](https://user-images.githubusercontent.com/10363029/212464308-63a844c9-5d85-4e59-94e3-2b91e516ba48.png)

I created this savegame right after I rebuild the bank in an attempt to remove the vans. Then I ran the simulation with and without TMPE. The bug is reproduced both time. I am now convinced it is a vanilla bug.
